File: Day 6/day6.ts
function inBox(box: string[]): boolean {

    if (!(box[0].replace('#','') === '' && box[box.length-1].replace('#','') === '')) return false

    for (const line of box.slice(1,-1)){
        if (line.slice(1,-1).includes('*')) return true
    }
    
    return false
}

File: Day 6/day6.py
def in_box(box:list[str]) -> bool:
  if not (box[0].replace('#','') == '' and box[-1].replace('#','') == ''): 
    return False

  for line in box[1:-1]:
    if '*' in line[1:-1]:
      return True
    
  return False